docs/hugo/content/design/ADR-2022-09-Multiple-Credential-Operator.md

```diff
@@ -0,0 +1,131 @@
+|--------------------|----------------------------------------------------------|----------------------------------------------------------------------|
+| Namespaced         | aso-credential                                           | Created in the Cluster namespace                                     |
+| Per-resource-group | serviceoperator.azure.com/credential-from: `SECRET_NAME` | Annotation on resource mentioning the secret to use for the resource |
+| Per-resource       | serviceoperator.azure.com/credential-from: `SECRET_NAME` | Annotation on resource mentioning the secret to use for the resource | 
+
+**Pros:**
+1. Users don't have to specify the configuration for each resource explicitly if using a namespaced secret
+2. Users get the flexibility of overriding the credential they want to use as per resource and as per resource group
+
+**Cons:**
+1. Could be complicated/confusing for users to understand the usage and credentials being used. This can be mitigated to an extent by having `serviceoperator.azure.com/credential-from` annoation on each resource using credential other than global.  
+
+### Option 4: Fixed secret name for namespace + configuration using global secret
+
+We'd use Fixed secret name for namespace + configuration using global secret for per-resource-group and per-resource secret. We can add another list(CREDENTIAL_FROM) in the global secret where admin users can specify the secrets they want to load on the startup.
+Then only these loaded secrets would be used further for per-resource and per-resource-group. This option would restrict the other non-admin users(who don't have access to the global secret) to add their credential. 
+
+**Pros:**
+1. Users don't have to specify the configuration for each resource explicitly if using a namespaced secret
+2. Security conscious, not anyone in the cluster can add to use their credentials.
+
+**Cons:**
+1. Could be complicated/confusing for users to understand the usage and credentials being used
+2. Not very flexible, as pod will have to startup again if a new secret needs to be loaded and not anyone can add secrets rather than the admin
+
+## Decision
+
+TBD
+
+## Handling for the failure cases and logging
+
+- Failure on loading a secret would result in failure on resource creation
+- Failure on authenticating the credentials would result in failure on resource creation
+- Logging should be clear about which credential is being used for a resource
+- Logging should be clear about the credential load/authentication failure
+- Errors can be reported through the standard `Condition` mechanism
+
+## Milestones
+
+1. Namespaced secrets: TBD
+2. per-ResourceGroup secrets: TBD
+3. per-Resource secrets: Will not be implemented unless there is strong user demand. We believe that per-ResourceGroup secrets should be sufficiently granular.
+
+## Open questions
+1. Do we need to support both per-resource and per-resource-group credentials? 
```

Comment:
I think you have actually answered this question now - we've designed for it so we _could_ support it, but not planned to implement unless user asks. To me, that means you can remove this from the list of open questions.